concent_api/core/payments/bankster.py

```diff
@@ -203,6 +203,7 @@ def finalize_payment(deposit_claim: DepositClaim) -> Optional[str]:
     """
 
     assert isinstance(deposit_claim, DepositClaim)
+    assert deposit_claim.tx_hash is None
```

Comment:
do we really want a crash here? What about logging an abnormal situation (I assume, this shouldn't normally happen)? Or even returning without a crash?

Reply:
> do we really want a crash here?

According to issue description, yes, this is abnormal situation. That also fits our coding style.

Reply:
We absolutely do not want to handle it (unless you consider returning HTTP 500 handling it). It's never supposed to happen. And the check is here mainly to protect programmers from making this mistake in the first place. As I said in the description of the issue - since this is an API it would make sense to raise an exeception too. But we should not handle that exception anyway.

If this happens at runtime - we want to bail out as soon as possible and get a report in Sentry. There's nothing we can do that does not involve fixing the code - and the server won't patch itself at runtime :)